gpkit/model.py

```diff
@@ -1,250 +1,424 @@
+        nu = result["sensitivities"]["monomials"]
+        nu_ = []
+        mons = 0
+        for mmap in mmaps:
+            max_idx = 0
+            for m_i in mmap:
+                if m_i is None:
+                    nu_.append(0)
                 else:
-                    removed_idxs.append(i)
-
-            k = []
-            count = 1
-            last = None
-            for p in p_idxs:
-                if last == p:
-                    count += 1
-                elif last is not None:
-                    k.append(count)
-                    count = 1
-                last = p
-            k.append(count)
-
-            varlocs, _ = locate_vars(exps)
-        else:
-            cs, exps, p_idxs, k, varlocs = (self.cs, self.exps, self.p_idxs,
-                                            self.k, self.varlocs)
-
-        missingbounds = {}
-        self.A = CootMatrix([], [], [])
-        for j, var in enumerate(varlocs):
-            varsign = "both" if "value" in var.descr else None
-            for i in varlocs[var]:
-                exp = exps[i][var]
-                self.A.append(i, j, exp)
-                if varsign is "both":
-                    pass
-                elif varsign is None:
-                    varsign = np.sign(exp)
-                elif np.sign(exp) != varsign:
-                    varsign = "both"
-
-            if varsign != "both" and var not in self.sweep:
-                if varsign == 1:
-                    bound = "lower"
-                elif varsign == -1:
-                    bound = "upper"
-                missingbounds[var] = bound
-
-        # add subbed-out monomials at the end
-        if not exps[-1]:
-            self.A.append(0, len(exps)-1, 0)
-        self.A.update_shape()
-
-        self.missingbounds = missingbounds
-        if printing:
-            self.checkbounds()
-
-        return cs, exps, self.A, p_idxs, k, removed_idxs
-
-    def checkbounds(self):
-        for var, bound in sorted(self.missingbounds.items()):
-            print("%s has no %s bound" % (var, bound))
+                    idx, percentage = m_i
+                    nu_.append(percentage*nu[idx + mons])
+                    if idx > max_idx:
+                        max_idx = idx
+            mons += 1 + max_idx
+        nu_ = np.array(nu_)
+        sensitivities["monomials"] = nu_
+
+        sens_vars = {var: sum([self.unsubbed_exps[i][var]*nu_[i]
+                               for i in locs])
+                     for (var, locs) in self.unsubbed_varlocs.items()}
+        sensitivities["variables"] = sens_vars
+
+        # free-variable sensitivities must be < arbitrary epsilon 1e-4
+        for var, S in sensitivities["variables"].items():
+            if var in freevariables and abs(S) > 1e-4:
+                print("free variable too sensitive:"
+                      " S_{%s} = %0.2e" % (var, S))
+
+        localexp = {var: S for (var, S) in sens_vars.items() if abs(S) >= 0.1}
```

Comment:
Not critical now, but eventually I'd vote for moving the `0.1` and the `1e-4` to kwarg defaults.